blocks/bricks/cost.py

```diff
@@ -58,7 +58,8 @@ def cost_matrix(self, y, y_hat):
 class CategoricalCrossEntropy(Cost):
     @application(outputs=["cost"])
     def apply(self, y, y_hat):
-        cost = tensor.nnet.categorical_crossentropy(y_hat, y).mean()
+        cost = tensor.nnet.categorical_crossentropy(y_hat, y).mean(
+            acc_dtype=floatX)
```

Comment:
I think it will be fixed with https://github.com/Theano/Theano/issues/2913.

I also checked that given the first `float32` and the second `int64` argument it returns `float32`, which is just what we need:

```
In [5]: tensor.nnet.categorical_crossentropy(numpy.array([[0.5, 0.5], [0.5, 0.5]], dtype='f32'), numpy.array([0, 1], dtype='i64')).eval()
Out[5]: array([ 0.69314718,  0.69314718], dtype=float32)
```